### PR TITLE
adding nexus go plugin impl

### PR DESCRIPTION
**Pull request by senthilhns**
Adding  Nexus artifact upload support in go for single and multi files

Pipeline execution link
https://app.harness.io/ng/account/gCoPSwHxS7ipOgx2iA9tOQ/all/orgs/default/projects/jenkinstoharness/pipelines/nexus_senthil_a4/executions/m_itUbaSSOy7rqkGZDTDHw/pipeline?storeType=INLINE

Screenshot
nxs

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM alpine:3.6 as alpine
```
Comment (Ompragash) [inline docker/Dockerfile]:
```suggestion
FROM alpine:3.20 as alpine
```

Can you please update all the alpine tag to 3.20 as older version contains some vulnerabilities?

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module github.com/harness-community/drone-nexus-publish
+
+go 1.12
```
Comment (Ompragash) [inline go.mod]:
```suggestion
go 1.22.7
```
also, always use Go 1.22.7 as the older versions got some vulns

**Review by Ompragash — CHANGES_REQUESTED** [review]

File: LICENSE.md
Comment (Ompragash) [inline LICENSE.md:1]: let's remove this file as all the plugins are Apache 2 and make it use the same - which I'll add once this PR is merged

File: docker/Dockerfile.linux.arm [hunk 3]
```diff
@@ -0,0 +1,10 @@
+FROM alpine:3.6 as alpine
```
Comment (Ompragash) [inline docker/Dockerfile.linux.arm]:
```suggestion
FROM alpine:3.20 as alpine
```

mentioned all the occurrences of alpine and go

File: docker/Dockerfile.linux.arm [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+RUN apk add -U --no-cache ca-certificates
+
+FROM alpine:3.6
```
Comment (Ompragash) [inline docker/Dockerfile.linux.arm]:
```suggestion
FROM alpine:3.20
```

File: docker/Dockerfile.linux.arm64 [hunk 5]
```diff
@@ -0,0 +1,10 @@
+FROM alpine:3.6 as alpine
```
Comment (Ompragash) [inline docker/Dockerfile.linux.arm64]:
```suggestion
FROM alpine:3.20 as alpine
```

File: docker/Dockerfile.linux.arm64 [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+RUN apk add -U --no-cache ca-certificates
+
+FROM alpine:3.6
```
Comment (Ompragash) [inline docker/Dockerfile.linux.arm64]:
```suggestion
FROM alpine:3.20
```

**Comment by Ompragash** [comment]
Please add windows support for the plugin ([LINK](https://github.com/drone-plugins/drone-s3/blob/master/docker/Dockerfile.windows.ltsc2022) to reference windows Dockerfile of plugins/s3)

File: plugin/nexus_plugin.go
Comment (Ompragash) [inline plugin/nexus_plugin.go:1]: as discussed on call, please move this plugin.go

**Review by Ompragash — COMMENTED** [review]
As discussed, keep all the plugin related code in plugin.go
- Dockerfiles should use alpine:3.20 for linux arcs
- after testing change the manifest.tmpl to refer images in plugins hub
- add tests for edge-case scenarios
    - IsMultiFileUploadArgsOk and IsSingleFileUploadArgsOk should validate for unexpected inputs
    - include scenarios where args.Attributes is invalid or missing values.

File: plugin/util.go [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+}
+
+func GetTestBuildRootDir() string {
+	return GetTestWorkSpaceDir()
+}
+
+const (
+	DefaultWorkSpaceDirEnvVarKey = "DRONE_WORKSPACE"
+	TestWorkSpaceDir             = "../test/tmp_workspace"
+)
```
Comment (Ompragash) [inline plugin/util.go]: As discussed last time, workspace path should not be hardcoded or treated as a dependency in the tests. Instead, tests should focus on covering the functionality and behavior of the plugin independently of specific environment setups. 

Consider abstracting workspace handling or mocking the relevant functions to ensure the tests are isolated and do not depend on specific paths

File: plugin/nexus_plugin.go
Comment (senthilhns) [reply plugin/nexus_plugin.go:1]: done

File: plugin/util.go (hunk 7 above)
Comment (senthilhns) [reply plugin/util.go]: The test cases rely on mock http client only, these were unused functions. Now they have all been removed.

File: docker/manifest.tmpl [hunk 8]
```diff
@@ -0,0 +1,31 @@
+image: senthilhns/drone-nexus-publish-single-store:{{#if build.tag}}{{trimPrefix "v" build.tag}}{{else}}latest{{/if}}
```
Comment (Ompragash) [inline docker/manifest.tmpl]:
```suggestion
image: plugins/nexus-publish:{{#if build.tag}}{{trimPrefix "v" build.tag}}{{else}}latest{{/if}}
```

File: docker/manifest.tmpl [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+{{#if build.tags}}
+tags:
+{{#each build.tags}}
+  - {{this}}
+{{/each}}
+{{/if}}
+manifests:
+  -
+    image: senthilhns/drone-nexus-publish-single-store:{{#if build.tag}}{{trimPrefix "v" build.tag}}-{{/if}}linux-amd64
```
Comment (Ompragash) [inline docker/manifest.tmpl]:
```suggestion
    image: plugins/nexus-publish:{{#if build.tag}}{{trimPrefix "v" build.tag}}-{{/if}}linux-amd64
```

File: docker/manifest.tmpl [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    platform:
+      architecture: amd64
+      os: linux
+  -
+    image: senthilhns/drone-nexus-publish-single-store:{{#if build.tag}}{{trimPrefix "v" build.tag}}-{{/if}}linux-arm64
```
Comment (Ompragash) [inline docker/manifest.tmpl]:
```suggestion
    image: plugins/nexus-publish:{{#if build.tag}}{{trimPrefix "v" build.tag}}-{{/if}}linux-arm64
```

File: docker/manifest.tmpl [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    platform:
+      variant: v8
+      architecture: arm64
+      os: linux
+  -
+    image: senthilhns/drone-nexus-publish-single-store:{{#if build.tag}}{{trimPrefix "v" build.tag}}-{{/if}}linux-arm
```
Comment (Ompragash) [inline docker/manifest.tmpl]:
```suggestion
    image: plugins/nexus-publish:{{#if build.tag}}{{trimPrefix "v" build.tag}}-{{/if}}linux-arm
```

File: docker/manifest.tmpl [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+    platform:
+      variant: v7
+      architecture: arm
+      os: linux
+  -
+    image: senthilhns/drone-nexus-publish-single-store:{{#if build.tag}}{{trimPrefix "v" build.tag}}-{{/if}}linux-arm
```
Comment (Ompragash) [inline docker/manifest.tmpl]:
```suggestion
    image: plugins/nexus-publish:{{#if build.tag}}{{trimPrefix "v" build.tag}}-{{/if}}linux-arm
```

File: README.md [hunk 13]
```diff
@@ -54,3 +54,53 @@ In Harness CI,
...
+
+In Harnes CI, YAML for multi file Upload
+```yaml
+- step:
+    type: Plugin
+    name: Plugin_1
+    identifier: Plugin_1
+    spec:
+      connectorRef: Docker_Hub_Anonymous
+      image: harnesscommunity/drone-nexus-publish:latest
```
Comment (Ompragash) [inline README.md]:
```suggestion
      image: plugins/nexus-publish:latest
```

File: README.md [hunk 14]
```diff
@@ -54,3 +54,53 @@ In Harness CI,
...
+    -e PLUGIN_USERNAME='nexususer01' \
+    -e PLUGIN_PASSWORD='some!secret@abc' \
+    -e PLUGIN_FORMAT='maven2' \
+    -e PLUGIN_GROUP_ID='test01' \
+    -e PLUGIN_PROTOCOL='http' \
+    -e PLUGIN_ARTIFACTS='[{"file": "file1.yaml", "classifier": "bin", "groupId": "test", "artifactId": "config-yaml-1", "type": "yaml", "version": "1"}, {"file": "file2.yaml", "classifier": "src", "groupId": "test", "artifactId": "all-config-yaml-2", "type": "yaml", "version": "2"}]' \
+    -e PLUGIN_REPOSITORY='stage-dev-repo' \
+      -v $(pwd):$(pwd) \
+      -w $(pwd) \
+    harnesscommunity/drone-nexus-publish
```
Comment (Ompragash) [inline README.md]:
```suggestion
    plugins/nexus-publish:latest
```

**Review by Ompragash — COMMENTED** [review]
also, just one small change: update go version to 1.22.7 and run `go mod tidy`

**Review by Ompragash — APPROVED** [review]
